lib/grape/validations/params_scope.rb

```diff
@@ -289,20 +289,22 @@ def validates(attrs, validations)
         opts = derive_validator_options(validations)
 
         # Validate for presence before any other validators
-        if validations.key?(:presence) && validations[:presence]
-          validate('presence', validations[:presence], attrs, doc_attrs, opts)
-          validations.delete(:presence)
-          validations.delete(:message) if validations.key?(:message)
-        end
+        validates_presence(validations, attrs, doc_attrs, opts)
 
         # Before we run the rest of the validators, let's handle
         # whatever coercion so that we are working with correctly
         # type casted values
         coerce_type validations, attrs, doc_attrs, opts
 
+        as = validations.delete(:as)
```

Comment:
The `validations` arg that is being passed into the function should not be modified to avoid side effects, make a copy.

Reply:
I refactored my changes to avoid mutating the arg however it is still being done within the method outside of my changes:
```ruby
 desc = validations.delete(:desc) || validations.delete(:description)
# ...
 doc_attrs[:documentation] = validations.delete(:documentation) if validations.key?(:documentation)
```
I'd like to try refactoring however the scope seems a little scary so perhaps as part of its own issue.

Reply:
Agreed. And thank you, do give it a shot.